### PR TITLE
Fix resolving `inherit_gem` in remote configs

### PR DESCRIPTION
**Pull request by CvX**
When a file inherits from a url inheriting from a gem, like so:

```yaml
# .rubocop.yml
inherit_from: http://example.com/default.yml
```

```yaml
# http://example.com/default.yml
inherit_gem:
    somegem: default.yml
```

```yaml
# somegem's default.yml
Layout/LineLength:
    Max: 48
```

Rubocop would then incorrectly try to resolve a URL like this:

`http://example.com//private/var/folders/69/bjrqv8v1225f7yqnxknm3fb80000gn/T/d20200518-94245-6dqeew/work/gems/somegem/default.yml`

With this change, local gem paths are no longer taken by mistake for relative remote paths.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/config_loader_resolver.rb [hunk 1]
```diff
@@ -172,6 +172,9 @@ def base_configs(path, inherit_from, file)
     def inherited_file(path, inherit_from, file)
       if remote_file?(inherit_from)
         RemoteConfig.new(inherit_from, File.dirname(path))
+      elsif Pathname.new(inherit_from).absolute?
```
Comment (bbatsov) [inline lib/rubocop/config_loader_resolver.rb:175]: I think we should probably add some comments for each branch of the `if`, otherwise it's getting really hard to follow which case maps to what exactly.

Comment (CvX) [reply lib/rubocop/config_loader_resolver.rb:176]: @bbatsov I've added the comments. Let me know if you see a way to improve them further.

Btw, the 2nd and 4th branch could be merged into `Pathname.new(inherit_from).absolute? || !file.is_a?(RemoteConfig)` but I think the way it's currently written is a bit more clear. And avoids unnecessary `File.expand_path` and `File.dirname` in the case of absolute paths.

Comment (bbatsov) [reply lib/rubocop/config_loader_resolver.rb:176]: Agreed.

**Comment by bbatsov** [comment]
Thanks!
